From 0298fcf59ec2ac833d1b45460a8ee2ccc892873d Mon Sep 17 00:00:00 2001
From: Socheat <github@socheat.dev>
Date: Sat, 27 Jun 2026 20:16:51 +0700
Subject: [PATCH 1/4] Refactor directory paths for module and resources

---
 core/src/core.cpp | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/core/src/core.cpp b/core/src/core.cpp
index 8984191c2..d235f421a 100644
--- a/core/src/core.cpp
+++ b/core/src/core.cpp
@@ -637,19 +637,8 @@ int sdrpp_main(int argc, char* argv[]) {
 #endif
     defConfig["transcieverLayout"] = 0;
 
-#if defined(_WIN32)
-    defConfig["modulesDirectory"] = "./modules";
-    defConfig["resourcesDirectory"] = "./res";
-#elif defined(IS_MACOS_BUNDLE)
-    defConfig["modulesDirectory"] = "../Plugins";
-    defConfig["resourcesDirectory"] = "../Resources";
-#elif defined(__ANDROID__)
-    defConfig["modulesDirectory"] = root + "/modules";
-    defConfig["resourcesDirectory"] = root + "/res";
-#else
-    defConfig["modulesDirectory"] = INSTALL_PREFIX "/lib/sdrpp/plugins";
-    defConfig["resourcesDirectory"] = INSTALL_PREFIX "/share/sdrpp";
-#endif
+    defConfig["modulesDirectory"] = "@prefix@/lib/sdrpp/plugins";
+    defConfig["resourcesDirectory"] = "@prefix@/share/sdrpp";
 
     // Load config
     std::error_code ec;

From 25987c2a8e1c316a5d859fd1fff053a9682a77b7 Mon Sep 17 00:00:00 2001
From: Socheat <github@socheat.dev>
Date: Sat, 27 Jun 2026 20:18:05 +0700
Subject: [PATCH 2/4] Implement transformPaths method in ConfigManager

Added a method to transform directory paths in configuration.
---
 core/src/config.cpp | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/core/src/config.cpp b/core/src/config.cpp
index 4d361ec2a..48ba5a018 100644
--- a/core/src/config.cpp
+++ b/core/src/config.cpp
@@ -108,3 +108,28 @@ void ConfigManager::autoSaveWorker() {
         }
     }
 }
+
+void ConfigManager::transformPaths(json& conf, bool expand) {
+    const char* env_prefix = std::getenv("SDRPP_PREFIX");
+    const std::string prefix = env_prefix ? env_prefix : INSTALL_PREFIX;
+
+    std::vector<std::string> directoryKeys = {
+        "modulesDirectory",
+        "resourcesDirectory"
+    };
+
+    const std::string from = expand ? "@prefix@" : prefix;
+    const std::string to = expand ? prefix : "@prefix@";
+
+    for (const auto& key : directoryKeys) {
+        if (conf.contains(key)) {
+            auto dir = conf[key].get<std::string>();
+
+            size_t pos = dir.find(from);
+            if (pos != std::string::npos) {
+                dir.replace(pos, from.length(), to);
+                conf[key] = dir;
+            }
+        }
+    }
+}

From 3766dcc074de1583e7875eb1c406250c48fcf9a3 Mon Sep 17 00:00:00 2001
From: Socheat <github@socheat.dev>
Date: Sat, 27 Jun 2026 20:19:02 +0700
Subject: [PATCH 3/4] Add transformPaths method declaration to config.h

---
 core/src/config.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/src/config.h b/core/src/config.h
index fbbdeb4a5..76dd86f98 100644
--- a/core/src/config.h
+++ b/core/src/config.h
@@ -33,4 +33,5 @@ class ConfigManager {
     std::mutex termMtx;
     std::condition_variable termCond;
     volatile bool termFlag = false;
-};
\ No newline at end of file
+    static void transformPaths(json& conf, bool expand);
+};

From 1e2490ad8215586c9be880ef446b102ec97d21a0 Mon Sep 17 00:00:00 2001
From: Socheat <github@socheat.dev>
Date: Sat, 27 Jun 2026 20:20:22 +0700
Subject: [PATCH 4/4] Transform paths in config before saving

Transform paths in configuration before saving.
---
 core/src/config.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/src/config.cpp b/core/src/config.cpp
index 48ba5a018..add02b69a 100644
--- a/core/src/config.cpp
+++ b/core/src/config.cpp
@@ -46,6 +46,9 @@ void ConfigManager::load(json def, bool lock) {
         conf = def;
         save(false);
     }
+
+    transformPaths(conf, true);
+
     if (lock) { mtx.unlock(); }
 }
 
@@ -54,7 +57,9 @@ void ConfigManager::save(bool lock) {
     auto justpath = wstr::str2wstr(path);
     auto newpath = wstr::str2wstr(path + ".new");
     std::ofstream file(newpath);
-    file << conf.dump(4);
+    auto tmp_conf = conf;
+    transformPaths(tmp_conf, false);
+    file << tmp_conf.dump(4);
     file.close();
     std::filesystem::rename(newpath, justpath);
     if (lock) { mtx.unlock(); }
